File: src/main/java/kr/co/lunatalk/domain/productlike/repository/ProductLikeRepositoryCustom.java
package kr.co.lunatalk.domain.productlike.repository;

import java.util.List;
import java.util.Map;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface ProductLikeRepositoryCustom {

	Long countByProductId(Long productId);

	Map<Long, Long> countByProductIds(List<Long> productIds);

	Map<Long, Boolean> existsByMemberIdAndProductIds(Long memberId, List<Long> productIds);

	Page<Long> findLikedProductIdsByMemberId(Long memberId, Pageable pageable);
}

File: src/main/kotlin/kr/co/lunatalk/domain/productlike/repository/ProductLikeRepositoryCustom.kt
package kr.co.lunatalk.domain.productlike.repository

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface ProductLikeRepositoryCustom {

    fun countByProductId(productId: Long): Long?

    fun countByProductIds(productIds: List<Long>): Map<Long, Long>

    fun existsByMemberIdAndProductIds(memberId: Long, productIds: List<Long>): Map<Long, Boolean>

    fun findLikedProductIdsByMemberId(memberId: Long, pageable: Pageable): Page<Long>
}
